frontend/src/hooks/useFilters.tsx

```diff
@@ -35,13 +35,11 @@ export const useFilters = () => {
       clearSearchParams('availabilityFromDate');
       clearSearchParams('availabilityToDate');
     } else {
-      setSearchUrlParams((prev: any) => ({
-        ...prev,
-        availabilityFromDate: range.from ? datePST(range.from, true) : undefined,
-        availabilityToDate: range?.to
-          ? datePST(range.to, true)
-          : range.from && datePST(range.from, true),
-      }));
+      searchParamsUrl.set('availabilityFromDate', datePST(range.from, true));
+      if (range.to) {
+        searchParamsUrl.set('availabilityToDate', datePST(range.to, true));
+      }
```

Comment:
Curious how searchParamsUrl.set behaves differently than setSearchParamsUrl? (Just wondering if we should switch to this elsewhere in the application?)